src/ServicePulse.Host/app/js/views/monitored-endpoints/monitored-endpoints.controller.js

```diff
@@ -13,12 +13,9 @@
         });
 
         function updateUI() {
-            monitoringService.getRaw().then(function (data) {
-                $scope.endpoints = data["NServiceBus.Endpoints"];
-
-                timeoutId = $timeout(function () {
-                    updateUI();
-                }, 5000);
+            monitoringService.getEndpoints().subscribe(function (endpoint) {
+                $scope.endpoints = $scope.endpoints || {};
+                $scope.endpoints[endpoint.endpointName] = endpoint;
```

Comment:
The map function in the service maps the array of URLs into http GETs, which are wrapped in observables each to be repeated every 5 seconds (and auto retried), and then merged into a single stream which pipes the individual results as a merged stream to this code